lib/modules/manager/custom/index.ts

```diff
@@ -0,0 +1,24 @@
+import { logger } from '../../../logger';
+import type { ExtractConfig, PackageFile, Result } from '../types';
+import managers from './api';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config: ExtractConfig
+): Result<PackageFile | null> {
+  const { customType = 'regex' } = config;
+  if (!managers.has(customType)) {
+    logger.warn({ customType }, 'Unknown customManagers type');
+    return null;
+  }
+
+  const m = managers.get(customType)!;
+  return m?.extractPackageFile
+    ? m.extractPackageFile(content, fileName, config)
+    : null;
```

Comment:
```suggestion
  return m.extractPackageFile!.(content, fileName, config);
```
we should use a custom manager  API interface and make that function mandatory.